### PR TITLE
[molecule-ansible] Add post for using molecule with ansible

### PR DESCRIPTION
**Pull request by dankolbrs**
Add post for using Molecule for Ansible role creation

In a comment on the pull request, include the following information and check off the list.

 - [x] Mini bio
  - Soft capped to 75 words
 - [x] Desired date of publishing
  - Make sure this matches the `date` in the front matter
 - [x] Social media contact info
 - [x] Redaction: (yes|no)
  - Allow redaction, editing or structure changes by technical writer staff
 - [x] Comments: (yes|no)
  - If you're willing to answer comments about your post, set `comments: true`.
  - Make sure to subscribe to your post comments by clicking on the **Subscribe** link at the very bottom of the post _once_ it's published.

@catlook Please review.

### CODE REVIEW

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-c7c75a1c90/)

**Comment by dankolbrs** [comment]
I put tomorrow in for the publish date, but will be stepping out early this afternoon. Despite my reviewing, I *know* there will be errors found :D. 

If I can't get to them in time and it needs to be pushed back to early next week, let me know and I'll edit!

**Comment by catlook** [comment]
Dan, I'll review...If I find anything, I'll tweak myself and publish today.  If you're here, I'll check with you first!

File: _posts/2018-09-28-molecule-for-ansible-role-creation.md [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+    changed: [localhost]
+    
+    PLAY RECAP *********************************************************************
+    localhost                  : ok=3    changed=2    unreachable=0    failed=0
+    
+    
+
+```
+
+Great! With all that, now we know that our Ansible and Python tests are linted, and our tests run meaning our role does what we **intend** for it to do.
```
Comment (catlook) [inline _posts/2018-09-28-molecule-for-ansible-role-creation.md]: ... tests run. This means our ...

File: _posts/2018-09-28-molecule-for-ansible-role-creation.md [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+
+### Molecule verify
+
+I did skip a step here. So far I have described the steps of:
```
Comment (catlook) [inline _posts/2018-09-28-molecule-for-ansible-role-creation.md]: ...  the steps for: ...

File: _posts/2018-09-28-molecule-for-ansible-role-creation.md [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+Lint completed successfully.
+--> Executing Flake8 on files found in /home/dan/Projects/example_playbooks/nginx_install/molecule/default/tests/...
+Lint completed successfully.
+--> Executing Ansible Lint on /home/dan/Projects/example_playbooks/nginx_install/molecule/default/playbook.yml...
+Lint completed successfully.
+```
+
+## Conclusion
+
+Molecule is a great abstraction for the multiple steps of create, test, clean process that happens during development of an Ansible role. Not only does it create and provide sane defaults to the directory structure of a role, it makes it easy to create a test a role during development. While there is a bit of a learning curve, the increased productivity of testing during development makes it an absolutely worthwhile investment.
```
Comment (catlook) [inline _posts/2018-09-28-molecule-for-ansible-role-creation.md]: ... test, and clean that happen during the development ...

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/build-b7b2de4114/)

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-e859119375/). :bow:

File: _posts/2018-09-28-molecule-for-ansible-role-creation.md [hunk 4]
```diff
@@ -0,0 +1,663 @@
...
+categories:
+    - Automation
+    - DevOps
+    - Ansible
+    - Orchestration
+---
+
+In our Quality Engineering organization, we create, configure, and destroy a lot of servers via automation. [Ansible](https://ansible.com) is a great method for handling the configuration of servers, but the creation of Ansible roles and playbooks can be trial and error for even experienced Operations Engineers. [Molecule](https://molecule.readthedocs.io/en/latest/) provides a way to speed up the development and confidence of Ansible roles and playbooks by wrapping a virtualization driver with tools for testing and linting.
+
+<!-- more -->
```
Comment (dankolbrs) [inline _posts/2018-09-28-molecule-for-ansible-role-creation.md]: @catlook ,
I want to make sure to draw attention to L18 and L20 to hopefully make it easier to review.

I finished the sentence this time:
```
... provides a way to speed up the development and confidence of Ansible roles and playbooks by wrapping a virtualization driver with tools for testing and linting.
```

And I added the `<!-- more  -->` as I realized it was missing on first commit.

**Comment by dankolbrs** [comment]
Just realized github was hiding a few comments. Working on another commit...

File: _posts/2018-09-28-molecule-for-ansible-role-creation.md [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+
+def test_nginx_running(host):
+    nginx_service = host.service('nginx')
+    assert nginx_service.is_running
+    assert nginx_service.is_enabled
+```
+
+## Running Molecule
+
+We've got our role written, and our tests. We could just run `molecule test` and work through all the steps. But, I prefer running `create`, `converge`, and `test` all seperately, and in that order. This seperates the various steps and makes any failures easier to track down.
```
[the remark this answers is not in the document]
Comment (dankolbrs) [reply _posts/2018-09-28-molecule-for-ansible-role-creation.md]: I knew I was doing it too. I swear I did a `ctrl+f` before the commit but still missed them! Next time I'm regex'ing it.

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-0da4bc2d7e/). :bow:

**Comment by dankolbrs** [comment]
@catlook ,
I want to make sure to draw attention to L18 and L20 to hopefully make it easier to review due to the multiple commits (sorry!).

I finished the sentence this time:
```
... provides a way to speed up the development and confidence of Ansible roles and playbooks by wrapping a virtualization driver with tools for testing and linting.
```

And I added the `<!-- more  -->` as I realized it was missing on first commit.

Diff in commit:
https://github.com/rackerlabs/docs-developer-blog/pull/382/commits/b4f6d0a498ea53c9776c5e60f1a067c2a55261f6#diff-45436623ea0336dbf205f342386a7434R18

**Comment by catlook** [comment]
nice catch on the <more> line!  The makes the preview segment on the landing page work!

**Pull request merged**